posthog/models/filters/mixins/property.py

```diff
@@ -51,6 +68,24 @@ def _parse_properties(self, properties: Optional[Any]) -> List[Property]:
             )
         return ret
 
+    def _parse_property_group(self, group: Optional[Dict]) -> PropertyGroup:
```

Comment:
@neilkakkar check out this format. I changed the parsing to rely on recursion through these two functions. it makes the typing simpler and not rely on this `Union[List[Property], List[PropertyGroup], PropertyGroup]` situation. I left some of the TODOs especially if typings get mixed up. lmk what you think

Reply:
I wasn't too happy with that, like this better!